src/parcsr_ls/par_mod_multi_interp_device.c

```diff
@@ -86,7 +86,11 @@ struct local_equal_plus_constant : public
 };
 
 /* transform from local C index to global C index */
+#if (defined(THRUST_VERSION) && THRUST_VERSION < 101000)
 struct globalC_functor : public thrust::unary_function<HYPRE_Int, HYPRE_BigInt>
+#else
+struct globalC_functor
```

Comment:
The thrust in cuda 11+ doesn't support `public thrust::unary_function` anymore?

Reply:
Hi Rui Peng, It's been deprecated:

```
path-to-hypre/src/utilities/_hypre_utilities.hpp:1538:57: warning: ‘template<class Argument, class Result> struct thrust::THRUST_200700_860_NS::unary_function’ is deprecated [-Wdeprecated-declarations]
 1538 | struct type_cast : public thrust::unary_function<T1, T2>
      |                                                         ^             
path-to-spack/cuda-12.8.0-hd4fbybxbgvypwjgbekpfccmbevzr623/targets/x86_64-linux/include/thrust/functional.h:85:23: note: declared here
   85 | struct THRUST_DEPRECATED unary_function
      |                       ^~~~~~~~~~~~~~
```